en/02_Developer_Guides/17_CLI/02_Hybrid_Commands.md

```diff
@@ -0,0 +1,172 @@
+summary: Code that can be run both from the command line and via HTTP requests
+icon: tasks-alt
+---
+
+# `PolyCommand` {#polycommand}
+
+Some code needs to be accessible both via HTTP requests over the web and via CLI in your terminal. Common examples of this are building the database and flushing cache.
+
+The [`PolyCommand`](api:SilverStripe\PolyExecution\PolyCommand) class provides a consistent API for getting input and providing output regardless of the context where the code is run. The API for this class is intentionally similar to the API for the `Command` class provided by `symfony/console`.
+
+> [!TIP]
+> See [the symfony/console input documentation](https://symfony.com/doc/current/console/input.html#using-command-options) for more specific details about using `InputInterface` and `InputOption`. Note that we use the [`getOptions()`](api:SilverStripe\PolyExecution\PolyCommand::getOptions()) method instead of `configure()` to return an array of `InputOption` objects. `PolyCommand` does not allow arguments.
+>
+> See [the symfony/console colouring documentation](https://symfony.com/doc/current/console/coloring.html) for information about styling output. Note that all output (including outputting for HTTP requests) as well as the command description and help info can use the `symfony/console` styling format.
+
+```php
+namespace App\Cli\Command;
+
+use SilverStripe\PolyExecution\PolyCommand;
+use SilverStripe\PolyExecution\PolyOutput;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+
+class MyPolyCommand extends PolyCommand
+{
+    protected static string $commandName = 'app:my-command';
+
+    protected string $title = 'My command';
+
+    protected static string $description = 'A command that does something';
+
+    public function run(InputInterface $input, PolyOutput $output): int
+    {
+        if ($input->getOption('do-action')) {
+            $output->writeln('Doing something...');
+        }
+        return Command::SUCCESS;
+    }
+
+    public function getOptions(): array
+    {
+        return [
+            new InputOption('do-action', null, InputOption::VALUE_NONE, 'do something specific'),
+        ];
+    }
+}
+```
+
+> [!NOTE]
+> You can also optionally implement the static [`getHelp()`](api:SilverStripe\PolyExecution\PolyCommand::getHelp()) method to provide additional context about the command. This helps to keep the description short.
+
+Once you've set up your command in PHP, you can add an HTTP route to is using the regular [`Director` routing rules](/developer_guides/controllers/routing/), and add it to Sake with the [`Sake.commands`](api:SilverStripe\Cli\Sake->commands) configuration property. Note that this isn't necessary for the special subclasses mentioned in the sections below.
+
+```yml
+---
+Name: polycommands
+---
+SilverStripe\Control\Director:
+  rules:
+    my-command: 'App\Cli\Command\MyPolyCommand'
```

Comment:
Is this now necessary? Or does protected static string $commandName provide a segment for dev/tasks the way that private static $segment used to?

Reply:
This is an optional way that custom poly commands can be routed for access via an HTTP request.
This is not necessary for `BuildTask` or `DevCommand`.